sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -380,6 +380,9 @@ object ColumnPruning extends Rule[LogicalPlan] {
         p
       }
 
+    // Eliminate the Projects with empty projectList
+    case p @ Project(projectList, child) if projectList.isEmpty => child
```

Comment:
I'm thinking of the correctness of this rule. Actually this is not column pruning, but add more columns, as `child` may have more one columns.

And why this rule `case p @ Project(projectList, child) if sameOutput(child.output, p.output) => child` can't work?